spec/syntax.md

```diff
@@ -653,10 +653,12 @@ While a reserved sequence is technically "well-formed",
 unrecognized _reserved-annotations_ or _private-use-annotations_ have no meaning.
 
 ```abnf
-reserved-annotation       = reserved-annotation-start reserved-body
+reserved-annotation       = reserved-annotation-start [[s] reserved-body]
```

Comment:
I think it's worth adding some prose here explaining why the grammar is structured this way. It wasn't obvious immediately that it had to be, so it won't be obvious to the reader either. Including the example would be good. 

I also think a note should be added to the "Reserved Statements" section of the same doc that explains what the limitations are of `reserved-statement` that arise because of this change. You could use your example from #721.

Reply:
Let's use a separate PR to do any text changes.